### ISSUE TITLE
[BUG] `no-curly-component-invocation` rule throwing unexpectedly

### ISSUE THREAD

**Issue by alexdiliberto** [ember-template-lint/ember-template-lint#857]
After enabling this rule, in my template files I'm seeing yielded arguments all throw the corresponding lint error

Example, from a template only glimmer-component:

```
{{#each @options as |option|}}
  {{option}}
{{/each}}
```

```
  2:2  error  You are using the component {{option}} with curly component syntax. You should use <Option> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['option'] }`.  no-curly-component-invocation
```

`{{option}}` isn't a component or helper, it's simply a POJO or some raw data passed into the component.

Is it really necessary to whitelist all yielded arguments like this?

---

Versions:

```
ember-template-lint@^1.1.0:
  version "1.5.3"
ember-cli-template-lint@^1.0.0-beta.3:
  version "1.0.0-beta.3"
```

**Comment by alexdiliberto** [comment]
some quick debugging, don't really have the time to dig in fully

```
node= { type: 'MustacheStatement',
  path:
   { type: 'PathExpression',
     original: 'option',
     this: false,
     parts: [ 'option' ],
     data: false,
     loc:
      SourceLocation { source: undefined, start: [Object], end: [Object] } },
  params: [],
  hash:
   { type: 'Hash',
     pairs: [],
     loc: { source: '(synthetic)', start: [Object], end: [Object] } },
  escaped: true,
  loc:
   { source: null,
     start: { line: 2, column: 2 },
     end: { line: 2, column: 12 } },
  strip: { open: false, close: false } }
entered block: "if (this.isDisallowed(name) || !this.config.requireDash) {"
this.config.requireDash= false
this.isDisallowed(name) false
message= You are using the component {{option}} with curly component syntax. You should use <Option> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['option'] }`.
```

**Comment by rwjblue** [comment]
This is a tricky one. For a block param, we can't _know_ for sure if it is a component invocation, some text, or heck even a DOM element. The rule does not make any different assumptions about yielded block params at the moment.

We could tweak things a bit so that the rule does not apply to block params when used without any arguments (e.g. `{{foo}}` vs either `{{foo whatever}}` or `{{foo some=thing}}`). 🤔 

@patocallaghan / @GavinJoyce / @Turbo87 / @bmish - What do y'all think?

**Comment by Turbo87** [comment]
due to the ambiguity of the curly invocation I'm not a big fan of the `no-curly-component-invocation` rule at all... :-/

but I agree that is should only warn if there are things passed into the component because otherwise it's impossible to know if it's a component or not

**Comment by alexdiliberto** [comment]
In my case, I actually don't have any single-name components so I can sidestep this by setting `'no-curly-component-invocation': { requireDash: true }` in the config...but obviously not an actual solution for everyone. Thanks for the quick feedback

**Comment by patocallaghan** [comment]
@alexdiliberto @rwjblue we had the same problem with regards the `requireDash: false` setting and we enable `requireDash: true` and then prevent any non-dashed components with the `disallow` list (we only have a very small number). 

`requireDash: true` is the default for the rule but it's turned to `requireDash: false` for the Octane config I believe.

But @rwjblue we could definitely make it smarter like you suggest. How easy is it to deduce whether something is an in-scope block param?

**Comment by rwjblue** [comment]
> How easy is it to deduce whether something is an in-scope block param?

Very :smile:

```js
MustacheStatement(node) {
  if (this.isLocal(node)) {
    return;
  }

  //...snip... 
}
```

**Comment by patocallaghan** [comment]
👍 I'll put together a fix

**Comment by rwjblue** [comment]
That would be awesome, thank you @patocallaghan!

**Comment by rwjblue** [comment]
Fixed by https://github.com/ember-template-lint/ember-template-lint/pull/861
